src/components/SlidingNavigation/SlidingNavigation.js

```diff
@@ -0,0 +1,113 @@
+import * as React from "react";
+import PropTypes from "prop-types";
+import classNames from "classnames";
+
+import "./SlidingNavigation.css";
+
+const SlidingNavigation = ({ data, options }) => {
+  const activeItemClassName = `SlidingNavigation__item--active`;
+  const observerOptions = options || {
+    threshold: 0.2,
+  };
+
+  React.useEffect(() => {
+    const observer = new IntersectionObserver((entries) => {
+      entries.forEach((entry) => {
+        if (
+          entry.isIntersecting &&
+          !window.location.hash.includes(entry.target.id)
+        ) {
+          window.history.replaceState({}, ``, `#${entry.target.id}`);
```

Comment:
Would be nice to have solution without any history modifications.

Reply:
Solution was recommended by @grinkus-adapt 
I'm open to any other suggestions